src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutPresenter.cs

```diff
@@ -1,13 +1,1008 @@
+				CycleFocus(true /* shouldCycleDown */, focusState);
+			}
+			else if (focusState == FocusState.Unfocused)
+			{
+				// A child element got focus, so make sure we keep m_iFocusedIndex in sync
+				// with it.
+				var focusedElement = FocusManager.GetFocusedElement() as DependencyObject;
+
+				// Since GotFocus is an async event, the focused element could be null if we got it
+				// after the popup closes, which clears focus.
+				if (focusedElement != null)
+				{
+					var focusedElementIndex = -1;
+					focusedElementIndex = IndexFromContainer(focusedElement);
+					if (focusedElementIndex != -1)
+					{
+						m_iFocusedIndex = focusedElementIndex;
+					}
+				}
+			}
+
+
+		}
+
+		void EnsureInitialFocusIndex()
+		{
+			if (m_iFocusedIndex == -1)
+			{
+				var focusedElement = FocusManager.GetFocusedElement();
+
+				if (this != focusedElement)
+				{
+					var menuItemsCount = 0;
+					IObservableVector<object> menuItems = Items;
+
+					menuItemsCount = Items.Count;
+
+					for (var i = 0; i < menuItemsCount; ++i)
+					{
+						var itemAsDependencyObject = Items[i] as DependencyObject;
+						MenuFlyoutItem menuItem;
+
+						menuItem = itemAsDependencyObject as MenuFlyoutItem;
+
+						if (menuItem == focusedElement)
+						{
+							m_iFocusedIndex = i;
+							break;
+						}
+					}
+
+					global::System.Diagnostics.Debug.Assert(m_iFocusedIndex != -1);
+				}
+			}
+		}
+
+		int GetPositionInSetHelper(MenuFlyoutItemBase item)
+		{
+			var returnValue = -1;
+
+			var presenter= item.GetParentMenuFlyoutPresenter();
+
+			if (presenter != null)
+			{
+				IObservableVector<object> items = presenter.Items;
+
+				var indexOfItem = 0;
+				indexOfItem = Items.IndexOf(item);
+
+				if (indexOfItem != -1)
+				{
+					// Iterate through the items preceding this item and subtract the number
+					// of separaters and collapsed items to get its position in the set.
+					var positionInSet = (int)(indexOfItem);
+
+					for (var i = 0; i < indexOfItem; ++i)
+					{
+						var child = Items[i] as DependencyObject;
+
+						if (child != null)
+						{
+							if (child is MenuFlyoutSeparator)
+							{
+								--positionInSet;
+							}
+							else
+							{
+								if (child is UIElement itemAsUI)
+								{
+									var visibility = Visibility.Collapsed;
+									visibility = itemAsUI.Visibility;
+
+									if (Visibility.Visible != visibility)
+									{
+										--positionInSet;
+									}
+								}
+							}
+						}
+					}
+
+					global::System.Diagnostics.Debug.Assert(positionInSet >= 0);
+
+					returnValue = positionInSet + 1;  // Add 1 to convert from a 0-based index to a 1-based index.
+				}
+			}
+
+			return returnValue;
+		}
+
+		int GetSizeOfSetHelper(MenuFlyoutItemBase item)
+		{
+			var returnValue = -1;
+
+			var presenter = item .GetParentMenuFlyoutPresenter();
+
+			if (presenter != null)
+			{
+				IObservableVector<object> items;
+				items = presenter.Items;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'items'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)